### DISCUSSION TITLE
Downsides to enabling debugging

### DISCUSSION THREAD

**Discussion by suetanvil** [mruby/mruby#5573] [Q&A] [answered]
What are the negatives to setting `conf.enable_debug` in my `build_config.rb`?

Specifically, if I use an `mruby` build with debugging enabled but I don't use debugging features (i.e. I've run `mrbc` without `-g`), does this impose any efficiency costs?

**Comment by matz** [comment, marked as answer]
`enable_debug` cost you:

* memory for keeping debug information (especially symbols and line numbers)
* code size (by enabling assertion checks)

But efficiency cost is small.

**Reply by suetanvil** [reply]
Thanks.  My use case is creating a binary distribution of the mruby tools.  It seems like including `mrdb` is worth the overhead.
